### PR TITLE
Fix events having access to past-gen exclusive moves

### PR DESCRIPTION
**Pull request by Karthik99999**
https://www.smogon.com/forums/threads/bug-reports-v4-read-original-post-before-posting.3663703/post-9373659

### CODE REVIEW

File: sim/team-validator.ts [hunk 1]
```diff
@@ -1767,6 +1775,12 @@ export class TeamValidator {
 		}
 		// Event-related ability restrictions only matter if we care about illegal abilities
 		const ruleTable = this.ruleTable;
+		if (ruleTable.has('obtainablemoves')) {
+			if (eventData.generation > setSources.maxSourceGen() && !(dex.gen <= 2 && eventSpecies.gen === 1)) {
+				if (fastReturn) return true;
+				problems.push(`${name} must not have moves only learnable in gen ${setSource.maxSourceGen()}${etc}.`);
```
Comment (Karthik99999) [inline sim/team-validator.ts:1781]: Would appreciate suggestions for this message

Comment (singiamtel) [reply sim/team-validator.ts]: Maybe just ``${name} does not learn ${move.name} in gen ${dex.gen}${etc}.``

File: sim/team-validator.ts [hunk 2]
```diff
@@ -1767,6 +1775,13 @@ export class TeamValidator {
 		}
 		// Event-related ability restrictions only matter if we care about illegal abilities
```
Comment (urkerab) [inline sim/team-validator.ts:1776]: This comment is a bit confusing now that you've added a check for illegal moves.

File: sim/team-validator.ts [hunk 3]
```diff
@@ -1767,6 +1775,13 @@ export class TeamValidator {
...
 		const ruleTable = this.ruleTable;
+		if (ruleTable.has('obtainablemoves')) {
+			const tradebackEligible = ruleTable.has('allowtradeback') || (dex.gen === 2 && species.gen === 1);
```
Comment (urkerab) [inline sim/team-validator.ts:1779]: The other uses of `ruleTable.has('allowtradeback')` don't do this so I assume that there is a reason this is done differently here.

Comment (Karthik99999) [reply sim/team-validator.ts:1779]: My logic was that if tradebacks were allowed then this issue of not being able to use moves from a past gen wouldn't be a problem. Though I guess that's under the assumption that `allowtradeback` was also meant for you to be able to bring the mon back up to the current gen, which I'm not sure is the case.

File: sim/team-validator.ts (hunk 2 above)
Comment (Karthik99999) [reply sim/team-validator.ts:1776]: is this comment even necessary?

**Review by Marty-D — APPROVED** [review]

File: sim/team-validator.ts (hunk 2 above)
Comment (DaWoblefet) [reply sim/team-validator.ts:1776]: Yeah, you can take it out.

File: sim/team-validator.ts (hunk 3 above)
Comment (DaWoblefet) [reply sim/team-validator.ts:1778]: I'm in agreement here with urkerab that it should be removed; that's more of a policy thing with allowtradeback in custom rules for modern gens.

**Comment by DaWoblefet** [comment]
Thanks Karthik!

**Pull request merged**